Made `_weighted_percentile` raise value error when weights are all zero. (#31041)
scikit-learn/scikit-learn: doc/whats_new/upcoming_changes/sklearn.utils/31041.fix.rst

> - In :mod:`utils.stats` updated _weighted_percentile raise an value error when all weights are zero.

Reviewer: ```suggestion
- Updated `utils.stats._weighted_percentile` to raise an error when all weights are zero.
```
Could you please also list all the downstream public classes/functions that this would affect?

Reviewer 2: I would rather not name private API in the changelog. Instead, identify how this code path can be reached from the public API and document the user-facing behavior change.

Reviewer: > If this code is not reachable from public API

It is reachable by public functions that use `_weighted_percentile`, and results in an incorrect result being given. e.g.:

```
In [8]: y_true = [3, -0.5, 2, 7]
   ...: 
   ...: y_pred = [2.5, 0.0, 2, 8]
   ...: 
   ...: median_absolute_error(y_true, y_pred, sample_weight=[0,0,0,0])
Out[8]: 1.0
```

Just for completeness I checked numpys quantile function:

```python
np.quantile([1,2,3,4], 0.5, weights=[0,0,0,0], method='inverted_cdf')
```
gives the error:
```
RuntimeWarning: invalid value encountered in divide
  cdf /= cdf[-1, ...]  # normalization to 1
```

Error message could be improved to something specific for this case, but at least an error is given.

>     Also changed test_weighted_percentile_zero_weight to check for that error.

Reviewer: This line is not needed.
